agent/src/balrogagent/cmd.py

```diff
@@ -7,53 +7,68 @@
+    for endpoint in V1_SCHEDULED_CHANGE_ENDPOINTS:
+        logging.debug("Looking for active scheduled changes for endpoint %s..." % endpoint)
+        resp = await client.request(balrog_api_root, "/scheduled_changes/%s" % endpoint, loop=loop, auth0_secrets=auth0_secrets)
+        sc = resp["scheduled_changes"]
+        if endpoint == "rules":
+            # Rules are sorted by priority, when available. Deletions will not have
+            # a priority set, so we treat them as the lowest priority possible.
+            sc = sorted(sc, key=lambda k: (k["when"], k["priority"] or 0), reverse=True)
+        logging.debug("Found %s", len(sc))
+        for change in sc:
+            logging.debug("Processing change %s", change["sc_id"])
+            ready = False
+
+            # Figure out if the change is ready, which is type-specific.
+            if change.get("telemetry_uptake"):
+                # TODO: maybe replace this with a simple client.request()...
+                current_uptake = await get_telemetry_uptake(change["telemetry_product"], change["telemetry_channel"], loop=loop)
+                ready = telemetry_is_ready(change, current_uptake)
+            elif change["when"]:
+                # "when" is to-the-millisecond timestamp that gets stored as an int.
+                # It needs to be converted back to a float before it can be compared
+                # against other timestamps.
+                ready = time_is_ready(change, time.time())
+            else:
+                logging.debug("Unknown change type!")
+
+            # Check if all the required signoffs have been obtained
+            required_signoffs = change.get("required_signoffs") or {}
+            signoffs = change.get("signoffs") or {}
+            if not verify_signoffs(required_signoffs, signoffs):
+                logging.debug("Signoff requirements unmet, marking as not ready")
+                ready = False
+
+            # If we have all required signoffs, go ahead
+            if ready:
+                logging.debug("Change %s is ready, enacting", change["sc_id"])
+                url = "/scheduled_changes/{}/{}/enact".format(endpoint, change["sc_id"])
+                await client.request(balrog_api_root, url, method="POST", auth0_secrets=auth0_secrets, loop=loop)
+            else:
+                logging.debug("Change %s is not ready", change["sc_id"])
+
+
+async def process_v2_scheduled_changes(loop, balrog_api_root, auth0_secrets):
+    for endpoint in V2_SCHEDULED_CHANGE_ENDPOINTS:
+        resp = await client.request(balrog_api_root, endpoint, loop=loop, auth0_secrets=auth0_secrets)
+        for r in filter(lambda r: len(r["scheduled_changes"]) > 0, resp["releases"]):
+            t = time.time()
+            if all([time_is_ready(sc, t) for sc in r["scheduled_changes"]]):
+                await client.request(balrog_api_root, f"{endpoint}/{r['name']}/enact", method="POST", auth0_secrets=auth0_secrets, loop=loop)
```

Comment:
I'd suggest adding some logging to this function so it's similar to the v1 code branch, unless that's more verbose than is truly useful.